controller_manager:
  ros__parameters:
    update_rate: 100  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    # gripper_controller:
    #   type: joint_trajectory_controller/JointTrajectoryController

    arm_controller:
      type: position_controllers/GripperActionController

# gripper_controller:
#   ros__parameters:
#     joints:
#       - right_gripper_finger_joint
#       - left_gripper_finger_joint
#     gains:
#       right_gripper_finger_joint: {p: 100, i: 1.0, d: 10.0, i_clamp: 1.0}
#       left_gripper_finger_joint: {p: 100, i: 1.0, d: 10.0, i_clamp: 1.0}
#     constraints:
#       goal_time: 3.0
#       right_gripper_finger_joint: {goal: 0.02}
#       left_gripper_finger_joint: {goal: 0.02}

arm_controller:
  ros__parameters:
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6
    gains:
      joint_1: {p: 100.0, i: 0.01, d: 1.0}
      joint_2: {p: 100.0, i: 0.01, d: 1.0}
      joint_3: {p: 100.0, i: 0.01, d: 1.0}
      joint_4: {p: 100.0, i: 0.01, d: 1.0}
      joint_5: {p: 100.0, i: 0.01, d: 1.0}
      joint_6: {p: 100.0, i: 0.01, d: 1.0}
    constraints:
      goal_time: 10.0